dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/report/AbstractMetricsReporter.java

```diff
@@ -157,19 +159,17 @@ public void refreshData() {
                     switch (sample.getType()) {
                         case GAUGE:
                             GaugeMetricSample gaugeSample = (GaugeMetricSample) sample;
-                            List<Tag> tags = new ArrayList<>();
-                            gaugeSample.getTags().forEach((k, v) -> {
-                                if (v == null) {
-                                    v = "";
-                                }
-
-                                tags.add(Tag.of(k, v));
-                            });
+                            List<Tag> tags = getTags(gaugeSample);
 
                             Gauge.builder(gaugeSample.getName(), gaugeSample.getValue(), gaugeSample.getApply())
                                 .description(gaugeSample.getDescription()).tags(tags).register(compositeRegistry);
                             break;
                         case COUNTER:
+                            CounterMetricSample counterMetricSample = (CounterMetricSample) sample;
+                            FunctionCounter.builder(counterMetricSample.getName(),  counterMetricSample.getValue(),
+                                    Number::doubleValue).description(counterMetricSample.getDescription())
+                                .tags(getTags(counterMetricSample))
+                                .register( MetricsGlobalRegistry.getCompositeRegistry());
```

Comment:
Use compositeRegistry to be consistent with the above

Reply:
A few minor issues, no issues with functionality

Reply:
> Use compositeRegistry to be consistent with the above

done